### ISSUE TITLE
Vague Unicode Character Class Descriptions in 101-bundle-json

### ISSUE THREAD

**Issue by astrieanna** [deislabs/cnab-spec#209]
There are two sections of 101-bundle-json that mention unicode characters.

[Dotted Names](https://github.com/deislabs/cnab-spec/blob/master/101-bundle-json.md#dotted-names):
> ...a dotted name, which is defined herein as a name composed by concatenating name components (**Unicode letters** and the dash (-) character) together, separated by dot (.) characters.

[Name and Version: Identifying Metadata](https://github.com/deislabs/cnab-spec/blob/master/101-bundle-json.md#name-and-version-identifying-metadata):
> name MUST contain only characters from the **Unicode graph characters**

I added emphasis to the relevant phrase from each quote. Both of these phrases seem casual and vague as descriptions of sets of unicode characters, so I think they should be changed. Based on the context and my understanding of unicode, a possible replacement would be:

* "unicode letters" becomes "any unicode code point with a General Category of Lu, Ll, Lt, Lm, or Lo (a General Major Category of Letter)"
* "unicode graph characters" becomes "Unicode code points with a basic type of Graphic"

The replacement for unicode letters seems a bit too wordy, but I'm not sure what to write that is explicit, precise, and google-able, but also doesn't use so many words.

In case my interpretation of dotted name is off (since all the examples use ASCII letters only), the above definition would include things like `𝕔𝕠𝕞.𝕖𝕩𝕒𝕞𝕡𝕝𝕖.𝕞𝕪𝕒𝕡𝕡.𝕡𝕠𝕣𝕥` and `snʇɐʇs.uoıʇɔɐ.ǝʃdɯɐxǝ.ƃɹo` because each character (other than dot) has a general category of Ll.

## Summary
Problem: the descriptions of unicode character sets make it hard to be certain precisely what is included or excluded.

Questions:
* Are my suggested replacements correct?
* Is there a better way to word the unicode letters replacement?

cc @ryanmoran

**Comment by technosophos** [comment]
My understanding is that both _unicode letters_  and _unicode graphic characters_ (not `graph`)  are acceptable technical descriptions for their respective groups of unicode characters. The Unicode spec is explicit about what _unicode graphic characters_ means (Easiest read: http://www.unicode.org/glossary/#graphic_character), so I am not sure that needs more definition. If you want to list the individual letter categories of the Letter general category, feel free... but I don't think it's a major point of confusion that we might only support, say, Lu when we say "Unicode letters". So feel free to write more technical descriptions if you'd like. How do other specs do this?

The intention of the Dotted Names section was to allow non-ASCII characters, including `snʇɐʇs.uoıʇɔɐ.ǝʃdɯɐxǝ.ƃɹo`. It irks me that so many specs still insist on ASCII characters simply because English speakers wrote the specs.

**Comment by astrieanna** [comment]
For graphic characters, I agree that it appears to be a typo (foiling my googling). I can make a PR to correct the typo and link to the glossary entry that you linked to above (for people unfamiliar with unicode).

For the letters, I noticed that the graphic character definition says "a character with the General Category of Letter (L)". I would be happy to use that rather than listing all of them; I'm not familiar with how character sets are usually referred to and over-fitted to the first reference table I found.

**Comment by technosophos** [comment]
That sounds excellent to me.

**Comment by ryanmoran** [comment]
Is it intentional that Dotted Names contain only Unicode Letters, dots, and dashes while the bundle `name` contain Graphic Characters, which is a much larger category of characters (Letters, Marks, Numbers, Punctuation, Symbols, and Spaces)?

**Comment by technosophos** [comment]
Yes -- dotted names are intended to be easy to parse, while the `name` field is designed for user display. And it was intended that dotted names could be turned into internationalized DNS names.
